### ISSUE TITLE
Add function-calc-no-invalid

### ISSUE THREAD

**Issue by ota-meshi** [stylelint/stylelint#3820]
> What is the problem you're trying to solve?

I want to check invalid expressions of the `calc()` function beforehand.

I think that this rule will solve issue #3806 as well.

> What solution would you like to see?

A new rule that reports an invalid expression of the `calc()` function.

I think that we can check it with `postcss-value-parser`.

:+1: Examples of **correct** css:

```css
.foo {width: calc(100% - 80px);}
```

:-1: Examples of **incorrect** css:

```css
.foo {width: calc();} /* empty expression */
.foo {width: calc(100% 80px);} /* missing operator */
.foo {width: calc(100% -80px);} /* missing operator */
.foo {width: calc(100% - - 80px);} /* unexpected operator */
.foo {width: calc(100% -);} /* unexpected operator */
.foo {width: calc(- 100%);} /* unexpected operator */
.foo {width: calc(100px + 80);} /* addition (`+`) with mixed `number` and `length` */
.foo {width: calc(100% + 80);} /* addition with mixed `number` and `percentage` */
.foo {width: calc(100px - 80);} /* subtraction (`-`) with mixed `number` and `length` */
.foo {width: calc(100px * 80px);} /* multiplication (`*`) by `length` */
.foo {width: calc(100 / 80%);} /* division (`/`) with mixed, and `number` on the left-hand side */
.foo {width: calc(100% / 0);} /* division by zero */
```

---
#### Additional context

https://www.w3.org/TR/css-values-3/#calc-notation
https://developer.mozilla.org/en-US/docs/Web/CSS/calc

**Comment by jeddy3** [comment]
The syntax is [quite complex](https://www.w3.org/TR/css-values-3/#calc-syntax). I wonder whether this is best handled via a parser with syntax validation, [like csstree](https://github.com/csstree/csstree).

Alternatively, this could start as your own experimental stylelint plugin. We can always move it into core  once we're confident it meets our [criteria for inclusion](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#criteria-for-inclusion).

**Comment by evilebottnawi** [comment]
Just information:
We have `calc` parser in https://github.com/postcss/postcss-calc (for minification), maybe we can move this to own package and reuse this package

**Comment by jeddy3** [comment]
> maybe we can move this to own package and reuse this package

I think it's definitely worth looking in to.

**Comment by ota-meshi** [comment]
@jeddy3 Thank you for your opinion.
@evilebottnawi Thank you for the information.

I implemented this. #3833
Can you give us further opinions?
